nautobot/ui/scripts/build.js

```diff
@@ -0,0 +1,224 @@
+            const publicPath = config.output.publicPath;
+            const buildFolder = path.relative(process.cwd(), paths.appBuild);
+            printHostingInstructions(
+                appPackage,
+                publicUrl,
+                publicPath,
+                buildFolder,
+                useYarn
+            );
+        },
+        (err) => {
+            const tscCompileOnError =
+                process.env.TSC_COMPILE_ON_ERROR === "true";
+            if (tscCompileOnError) {
+                console.log(
+                    chalk.yellow(
+                        "Compiled with the following type errors (you may want to check these before deploying your app):\n"
+                    )
+                );
+                printBuildError(err);
+            } else {
+                console.log(chalk.red("Failed to compile.\n"));
+                printBuildError(err);
+                process.exit(1);
+            }
+        }
+    )
+    .catch((err) => {
+        if (err && err.message) {
+            console.log(err.message);
+        }
+        process.exit(1);
+    });
+
+// Create the production build and print the deployment instructions.
+function build(previousFileSizes) {
+    console.log("Creating an optimized production build...");
+
+    const compiler = webpack(config);
+    return new Promise((resolve, reject) => {
+        compiler.run((err, stats) => {
+            let messages;
+            if (err) {
+                if (!err.message) {
+                    return reject(err);
+                }
+
+                let errMessage = err.message;
+
+                // Add additional information for postcss errors
+                if (Object.prototype.hasOwnProperty.call(err, "postcssNode")) {
+                    errMessage +=
+                        "\nCompileError: Begins at CSS selector " +
+                        err["postcssNode"].selector;
+                }
+
+                messages = formatWebpackMessages({
+                    errors: [errMessage],
+                    warnings: [],
+                });
+            } else {
+                messages = formatWebpackMessages(
+                    stats.toJson({ all: false, warnings: true, errors: true })
+                );
+            }
+            if (messages.errors.length) {
+                // Only keep the first error. Others are often indicative
+                // of the same problem, but confuse the reader with noise.
+                if (messages.errors.length > 1) {
+                    messages.errors.length = 1;
+                }
+                return reject(new Error(messages.errors.join("\n\n")));
+            }
+            if (
+                process.env.CI &&
+                (typeof process.env.CI !== "string" ||
+                    process.env.CI.toLowerCase() !== "false") &&
+                messages.warnings.length
+            ) {
+                // Ignore sourcemap warnings in CI builds. See #8227 for more info.
```

Comment:
Might want to look into directly running `npm build` in CI so that we get these warnings-as-errors checks?